### PR TITLE
feat!: update to support Noir 0.37.0

### PR DESCRIPTION
**Pull request by kashbrti**
# Description
fixed the issues with the new release
## Problem\*

Resolves 

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kashbrti** [comment]
I'm failing to run the formatter. Here's the error I'm getting. It's a bit weird that it panics because I can compile and also the tests pass. 

```
Message:  Expected token Semicolon, got: LeftBracket
Location: tooling/nargo_fmt/src/formatter.rs:194

This is a bug. We may have already fixed this in newer versions of Nargo so try searching for similar issues at https://github.com/noir-lang/noir/issues/.
If there isn't an open issue for this bug, consider opening one at https://github.com/noir-lang/noir/issues/new?labels=bug&template=bug_report.yml
```

**Comment by TomAFrench** [comment]
@asterite looks like there's still issues with the formatter

File: .vscode/settings.json
Comment (TomAFrench) [inline .vscode/settings.json:1]: This should be removed.

File: .vscode/launch.json
Comment (TomAFrench) [inline .vscode/launch.json:1]: Not sure what this is but I don't think it should be committed

**Comment by asterite** [comment]
Right, I made some changes to the formatter after 0.36.0 was released because otherwise it destroyed all the formatting that exists in the array literals here.

**Comment by kashbrti** [comment]
I think it works with nightly just not with 0.36.0. should we just go ahead and merge this?

**Comment by TomAFrench** [comment]
It'll need the formatter to be disabled first if we're going to merge without it.

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**